### ISSUE TITLE
Altering data with slice

### ISSUE THREAD

**Issue by mvecsei** [fusion-flap/flap#32]
It would be useful for me to be able to change the values of dataobject.data corresponding to a given coordinate value. E.g. to only change the data at 'Time' = t, while otherwise leaving the dataobject intact. This could probably be done with a modified version of the slice method.

I am assigning myself, since I wanted to do this for some time, but fell free to implement it. :)

**Comment by sandorzoletnik** [comment]
Maybe one could define the [] operator the flap.DataObject. In this case we could write:

a[{'Time':flap.Intervals(3,4)}] = 1

Would this be OK? The solution would be to use the internal functions in slice_data which generate the data indices and use them to address the  data matrix. I guess you would also like to do this with the errors? I don't know how to handle it.

**Comment by mvecsei** [comment]
I agree that sounds like a reasonable syntax. I will try to come up with something for the error.
